packages/gatsby/src/redux/actions/public.js

```diff
@@ -429,9 +430,9 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
 
   const oldPage: Page = store.getState().pages.get(internalPage.path)
   const contextModified =
-    !!oldPage && !_.isEqual(oldPage.context, internalPage.context)
+    !!oldPage && !deepEqual(oldPage.context, internalPage.context)
```

Comment:
This deep equals library was around 33% faster in my testing https://github.com/planttheidea/fast-equals